spacy/ml/tb_framework.pyx

```diff
@@ -449,3 +468,174 @@ def predict(ids, tokvecs):
+            sizes.states)
+            for i in range(sizes.states):
+                if not states[i].is_final():
+                    unfinished.push_back(states[i])
+            for i in range(unfinished.size()):
+                states[i] = unfinished[i]
+        sizes.states = unfinished.size()
+        scores.append(step_scores)
+        unfinished.clear()
+    free_activations(&activations)
+
+    return scores
+
+
+cdef WeightsC get_c_weights(model, const float* feats, np.ndarray[np.npy_bool, ndim=1] seen_mask) except *:
+    cdef np.ndarray lower_bias = model.get_param("lower_b")
+    cdef np.ndarray upper_W = model.get_param("upper_W")
+    cdef np.ndarray upper_b = model.get_param("upper_b")
+
+    cdef WeightsC output
+    output.feat_weights = feats
+    output.feat_bias = <const float*>lower_bias.data
+    output.hidden_weights = <const float *> upper_W.data
+    output.hidden_bias = <const float *> upper_b.data
+    output.seen_mask = <const int8_t*> seen_mask.data
+
+    return output
+
+
+cdef SizesC get_c_sizes(model, int batch_size, int tokens) except *:
+    cdef SizesC output
+    output.states = batch_size
+    output.classes = model.get_dim("nO")
+    output.hiddens = model.get_dim("nH")
+    output.pieces = model.get_dim("nP")
+    output.feats = model.get_dim("nF")
+    output.embed_width = model.get_dim("nI")
+    output.tokens = tokens
+    return output
+
+
+cdef ActivationsC alloc_activations(SizesC n) nogil:
+    cdef ActivationsC A
+    memset(&A, 0, sizeof(A))
+    resize_activations(&A, n)
+    return A
+
+
+cdef void free_activations(const ActivationsC* A) nogil:
+    free(A.token_ids)
+    free(A.unmaxed)
+    free(A.hiddens)
+    free(A.is_valid)
+
+
+cdef void resize_activations(ActivationsC* A, SizesC n) nogil:
+    if n.states <= A._max_size:
+        A._curr_size = n.states
+        return
+    if A._max_size == 0:
+        A.token_ids = <int*>calloc(n.states * n.feats, sizeof(A.token_ids[0]))
+        A.unmaxed = <float*>calloc(n.states * n.hiddens * n.pieces, sizeof(A.unmaxed[0]))
+        A.hiddens = <float*>calloc(n.states * n.hiddens, sizeof(A.hiddens[0]))
+        A.is_valid = <int*>calloc(n.states * n.classes, sizeof(A.is_valid[0]))
+        A._max_size = n.states
+    else:
+        A.token_ids = <int*>realloc(A.token_ids,
+            n.states * n.feats * sizeof(A.token_ids[0]))
+        A.unmaxed = <float*>realloc(A.unmaxed,
+            n.states * n.hiddens * n.pieces * sizeof(A.unmaxed[0]))
+        A.hiddens = <float*>realloc(A.hiddens,
+            n.states * n.hiddens * sizeof(A.hiddens[0]))
+        A.is_valid = <int*>realloc(A.is_valid,
+            n.states * n.classes * sizeof(A.is_valid[0]))
+        A._max_size = n.states
+    A._curr_size = n.states
+
+
+cdef void predict_states(CBlas cblas, ActivationsC* A, float* scores, StateC** states, const WeightsC* W, SizesC n) nogil:
+    cdef double one = 1.0
```

Comment:
`one` seems to be unused, and I don't think there's a disadvantage to just putting `1.0` in the `cblas.sgemm` call, so we can just remove the definition here.

Reply:
Checked the `.cpp` to be sure, the literal `1.0` ends up being a constant.